src/components/dataItem/filter/FilterSelect.js

```diff
@@ -43,15 +42,9 @@ const getOperators = (valueType, optionSet) => {
     return operators
 }
 
-const FilterSelect = ({
-    valueType,
-    filter,
-    optionSet,
-    optionSets,
-    onChange,
-    loadOptionSet,
-}) => {
-    const operators = getOperators(valueType, optionSet)
+const FilterSelect = ({ valueType, filter, optionSetId, onChange }) => {
+    const { optionSet, fetchOptionSet } = useOptionSet()
```

Comment:
It might be possible to write it similar to the useProgramStageDataElements hook, where you just pass the id as an argument to the hook, and then you can remove the useEffect on line 59 of this file.

Reply:
Fixed!